### DISCUSSION TITLE
ESPHome 2024 06 update YAML config issue....

### DISCUSSION THREAD

**Discussion by BluThunder2k** [emporia-vue-local/esphome#292] [Q&A]
I updated to ESPHome 6 and when you add the ota:   platform: esphome line - numerous other lines in the config error out - i've tried removing and changing stuff but I think I found a bug in ESPhome - anyone else seeing similar behavior?

**Comment by BJamin99** [comment]
I just ran the update and had to add that line, but it worked for me.

**Reply by beardymarrow** [reply]
Me too. I updated as per this - https://next.esphome.io/changelog/2024.6.0.html#ota-platforms

so, I changed
```
ota:
  password: "xxxx"
```

to

```
ota:
  platform: esphome
  password: "xxxx"
```
  
  and it all works fine now.

**Comment by BluThunder2k** [comment]
Interesting.   I add the line and things under it start having the red tilda's under the text when it comes to button, binary sensors, etc.   Tells me when I hover over it:

  
```
  Component not found: button.
  - platform: restart
    name: ESP Emporia Vue Restart
```

Hmm.    Syntax is correct on everything, and it compiled just fine on 2024-05-05 esphome.

**Comment by BJamin99** [comment]
Not sure where there is anything about "restart" in the config.

**Reply by BluThunder2k** [reply]
It was added by me, but even going stock it gives me the error...

**Comment by BluThunder2k** [comment]
Well think I have figured it out.   I can take my normal running yaml config and create a new ESP home device - paste that config in and it's OK works just fine.    Exact same config on my normal VUE device gives me the errors.   So not a Vue issue, but definitely an odd ESPHome issue.
